BackendServer/Models/ServerMessage.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace BackendServer.Models
+{
+    enum MessageType
+    {
+        Info,
+        Warn,
+        Error
+    }
+
+    internal class ServerMessage(MessageType messageType, string content) : Message(content)
```

Comment:
It would be nice if ServerMessage can self determine its message type from just the content and it should accept a type Message not content.

Reply:
How can we get the ServerMessage to determine its message type?
- Having a `TYPE` parameter in the request? Works but needs some protocol changes
- Having the server prepend a `INFO:` or `WARN:` at the beginning of the message content? Seems simple enough
---
I don't understand how passing a variable of type Message to the constructor of a class that inherits Message would help 🤔

Reply:
The "MessageType" should be renamed to Command type and it should list the different types of command we will support. 
![image](https://github.com/Ldev-The-Remaster/ChattingApp/assets/106236056/3c526da0-59a4-42e6-9b0b-de1179e629d8)

Reply:
I agree the mechanism of transforming a Message into a ServerMessage doesn't seem clear yet. The idea is to take raw data and handle it then filter it into its correct type. However you want to do this as long as it matches the design and works go for whatever you feel like works best

Reply:
Also Akrum is right, about the MessageType. The message type is either ServerMessage or TextMessage. Maybe some  renaming would be nice, Message->RawMessage, ServerMessage -> ServerCommand and text message is okay.